theano/compile/mode.py

```diff
@@ -170,9 +170,14 @@ class AddNoOutputFromInplace(gof.Optimizer):
     outputs
 
     """
+
+    def __init__(self, first_output_idx=0, last_output_idx=None):
+        self.feature = gof.NoOutputFromInplace(first_output_idx,
```

Comment:
What about a quick generalization of this. Make this opt take the feature as input and rename it to AddFeatuerOptimizer?